Minimum rust version

I had rust(c) version 1.36.0, and got this error with `cargo build`:

```
   Compiling bincode v1.2.1
   Compiling toml v0.5.6
   Compiling xi-rope v0.3.0 (/home/hoijui/src/xi-editor/rust/rope)
error[E0425]: cannot find function `take` in module `mem`
   --> rope/src/breaks.rs:228:29
    |
228 |             let leaf = mem::take(&mut self.leaf);
    |                             ^^^^ not found in `mem`

error[E0425]: cannot find function `take` in module `mem`
   --> rope/src/spans.rs:130:33
    |
130 |             let mut leaf = mem::take(&mut self.leaf);
    |                                 ^^^^ not found in `mem`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0425`.
error: Could not compile `xi-rope`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

I found out that `mem::take` was added in Rust 1.40.0.
In order for others to not have to figure this out themselfs,
I wanted to note this minimum rust version in the _Cargo.toml_.
[The option to do this (`package.rust`)](https://github.com/rust-lang/rfcs/pull/2495) is very new though,
and thus my old cargo did not yet support it, so the error stayed hte same. ;-)

I wonder if it might still make sense to put this in:

```
[package]
rust = "1.40"
```
?

Feel free to just close the issue if not.

--- a/rust/Cargo.toml
+++ b/rust/Cargo.toml
@@ -7,6 +7,7 @@ description = "Main process for xi-core, based on json-rpc"
 categories = ["text-editors"]
 repository = "https://github.com/xi-editor/xi-editor"
 edition = '2018'
+rust = "1.40"
 
 [dependencies]
 serde = "1.0"

